presto-accumulo/src/main/java/com/facebook/presto/accumulo/AccumuloMetadata.java

```diff
@@ -0,0 +1,427 @@
+        return handle;
+    }
+
+    @Override
+    public void finishInsert(ConnectorSession session, ConnectorInsertTableHandle insertHandle, Collection<Slice> fragments)
+    {
+        clearRollback();
+    }
+
+    private void rollbackInsert(ConnectorInsertTableHandle insertHandle)
+    {
+        // Rollbacks for inserts are off the table when it comes to data in Accumulo.
+        // When a batch of Mutations fails to be inserted, the general strategy
+        // is to run the insert operation again until it is successful
+        // Any mutations that were successfully written will be overwritten
+        // with the same values, so that isn't a problem.
+        throw new PrestoException(NOT_SUPPORTED, "Unable to rollback insert. Some rows may have been written. Please run your insert again.");
+    }
+
+    @Override
+    public ConnectorTableHandle getTableHandle(ConnectorSession session, SchemaTableName stName)
+    {
+        if (!listSchemaNames(session).contains(stName.getSchemaName().toLowerCase())) {
+            return null;
+        }
+
+        // Need to validate that SchemaTableName is a table
+        if (!this.listViews(session, stName.getSchemaName()).contains(stName)) {
+            AccumuloTable table = client.getTable(stName);
+            if (table == null) {
+                return null;
+            }
+
+            return new AccumuloTableHandle(connectorId, table.getSchema(), table.getTable(),
+                    table.getRowId(), table.isExternal(), table.getSerializerClassName(),
+                    table.getScanAuthorizations());
+        }
+
+        return null;
+    }
+
+    @Override
+    public List<ConnectorTableLayoutResult> getTableLayouts(ConnectorSession session,
+            ConnectorTableHandle table, Constraint<ColumnHandle> constraint,
+            Optional<Set<ColumnHandle>> desiredColumns)
+    {
+        AccumuloTableHandle tableHandle = checkType(table, AccumuloTableHandle.class, "table");
+        ConnectorTableLayout layout = new ConnectorTableLayout(
+                new AccumuloTableLayoutHandle(tableHandle, constraint.getSummary()));
+        return ImmutableList.of(new ConnectorTableLayoutResult(layout, constraint.getSummary()));
+    }
+
+    @Override
+    public ConnectorTableLayout getTableLayout(ConnectorSession session,
+            ConnectorTableLayoutHandle handle)
+    {
+        return new ConnectorTableLayout(
+                checkType(handle, AccumuloTableLayoutHandle.class, "layout"));
+    }
+
+    @Override
+    public ConnectorTableMetadata getTableMetadata(ConnectorSession session,
+            ConnectorTableHandle table)
+    {
+        AccumuloTableHandle tHandle = checkType(table, AccumuloTableHandle.class, "table");
+        checkArgument(tHandle.getConnectorId().equals(connectorId),
+                "table is not for this connector");
+        return getTableMetadata(new SchemaTableName(tHandle.getSchema(), tHandle.getTable()));
```

Comment:
You cannot return `null` here. `MetadataManager` feeds `TableMetadata` with `TableMetadata` which requires it to not be `null`.